src/Symfony/Component/Encryption/Ciphertext.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption;
+
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\MalformedCipherException;
+
+/**
+ * This class is responsible over the payload API.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.3
+ *
+ * @internal
+ */
+class Ciphertext
+{
+    /**
+     * @var string algorithm for the encryption
+     */
+    private $algorithm;
+
+    /**
+     * @var string the encoded payload
+     */
+    private $payload;
+
+    /**
+     * @var string nonce for the algorithm
+     */
+    private $nonce;
+
+    /**
+     * @var array<string, string> additional headers
+     */
+    private $headers = [];
+
+    private function __construct()
+    {
+    }
+
+    public static function create(string $algorithm, string $ciphertext, string $nonce, array $headers = []): self
+    {
+        $model = new self();
+        $model->algorithm = $algorithm;
+        $model->payload = $ciphertext;
+        $model->nonce = $nonce;
+        $model->headers = $headers;
+
+        return $model;
+    }
+
+    /**
+     * Take a string representation of the chiphertext and parse it into an object.
+     *
+     * @throws MalformedCipherException
+     */
+    public static function parse(string $input): self
+    {
+        $parts = explode('.', $input);
+        if (false === $parts || 4 !== \count($parts)) {
```

Comment:
false will never be returned here?
`If separator is an empty string (""), explode() will return false.`

Reply:
THank you